### DISCUSSION TITLE
How to use csv stored locally in Grafana

### DISCUSSION THREAD

**Discussion by adriantaw** [grafana/grafana-infinity-datasource#54] [Q&A]
Hi,
Thank you for developing this cool plugin!
I am in the middle of trying this plugin with my Grafana, running locally.
When I am trying to access a CSV file located locally inside of my PC, I always get this error:

> Error
> <!DOCTYPE html> <html lang="en"> <head> <script nonce=""> !(function () { if ('PerformanceLongTaskTiming' in window) { var g = (window.__tti = { e: [] }); g.o = new PerformanceObserver(function (l) { g.e = g.e.concat(l.getEntries()); }); g.o.observe({ entryTypes: ['longtask'] }); } })(); </script> <meta charset="utf-8" /> <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" /> <meta name="viewport" content="width=device-width" /> <meta name="theme-color" content="#000" /> <title>Grafana</title> <base href="/grafana/" /> <link rel="preload" href="public/fonts/roboto/RxZJdnzeo3R5zSexge8UUVtXRa8TVwTICgirnJhmVJw.woff2" as="font" crossorigin /> <link rel="icon" type="image/png" href="public/img/fav32.png" /> <link rel="apple-touch-icon" sizes="180x180" href="public/img/apple-touch-icon.png" /> <link rel="mask-icon" href="public/img/grafana_mask_icon.svg" color="#F05A28" /> <link rel="stylesheet" href="public/build/grafana.dark.854a662a71a515996d8d.css" /> <script nonce=""> performance.mark('css done blocking'); </script> <meta name="apple-mobile-web-app-capable" content="yes" /> <meta name="apple-mobile-web-app-status-bar-style" content="black" /> <meta name="msapplication-TileColor" content="#2b5797" /> <meta name="msapplication-config" content="public/img/browserconfig.xml" /> </head> <body class="theme-dark app-grafana"> <style> .preloader { height: 100%; flex-direction: column; display: flex; justify-content: center; align-items: center; } .preloader__enter { opacity: 0; animation-name: preloader-fade-in; animation-iteration-count: 1; animation-duration: 0.9s; animation-delay: 1.35s; animation-fill-mode: forwards; } .preloader__bounce { text-align: center; animation-name: preloader-bounce; animation-duration: 0.9s; animation-iteration-count: infinite; } .preloader__logo { display: inline-block; animation-name: preloader-squash; animation-duration: 0.9s; animation-iteration-count: infinite; width: 60px; height: 60px; background-repeat: no-repeat; background-size: contain; background-image: url("data:image/svg+xml,%3csvg version='1.1' id='Layer_1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' x='0px' y='0px' width='351px' height='365px' viewBox='0 0 351 365' style='enable-background:new 0 0 351 365%3b' xml:space='preserve'%3e %3cstyle type='text/css'%3e .st0%7bfill:url(%23SVGID_1_)%3b%7d %3c/style%3e %3cg id='Layer_1_1_'%3e

Any idea how to enter the address of my local file?
Note I dockerized my application and with Grafana I am trying to access a csv file inside of /data/csv directory

**Comment by yesoreyeram** [comment]
Currently, it is not supported. Either your file should be inline or served via valid http url.

**Comment by ashuvaid** [comment]
I am using a nginx docker container to host the csv files so as to get a http url to access that file.
I have a separate grafana docker container also running where i have used the Infinity plugin.
In the url section i specify the url http://<container-ip>:<port>/<filename>.csv. But the plugin throws error. Error snippet attached for reference
![Capture](https://user-images.githubusercontent.com/67621435/120770421-be37c380-c53b-11eb-95c3-6a8d7dde0c6e.JPG)

Request to please suggest way to overcome this error.

**Reply by yesoreyeram** [reply]
If grafana and csv are in different containers, you need to give the address of CSV container. ( the URL should be accessible inside grafana container ). Example: using something like `http://csv-container-name:8081/os/processingunit.csv` may help.

**Reply by ashuvaid** [reply]
Hi - Thanks for your prompt answer. I tried with the suggestion to use the csv-container-name. But still no luck. Am i missing something obivious ?
![Capture](https://user-images.githubusercontent.com/67621435/120786392-ca2b8180-c54b-11eb-8a41-bd505630cfd0.JPG)

**Reply by ashuvaid** [reply]
Update --

1) My running docker containers is as given below

$ docker ps
CONTAINER ID   IMAGE              COMMAND                  CREATED         STATUS         PORTS                              NAMES
8b1a37f7e1d9   ubuntu_grafana:3   "/bin/bash"              3 minutes ago   Up 3 minutes   0.0.0.0:3000->3000/tcp, 3031/tcp   forgrafana
c047b71a0d80   nginx:latest       "/docker-entrypoint.…"   3 hours ago     Up 3 hours     80/tcp, 0.0.0.0:8081->8081/tcp     ngnix_file_server


2) In the forgrafana container I tried to check reachability to the ngnix_file_server using the curl commands for index.html and the csv file as well. And both are found to be working. However on port 80 and not 8081.

root@8b1a37f7e1d9:/# curl http://ngnix_file_server:80/OSProcessingUnit.csv | head -3
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0beginTime,elementType,NodeName,ldn,duration,CPULoadTotal,MemPercentUsed
2021-03-22 22:00:00,ERIC-SAPC,mpsapc1,PL-11,900,12,24
2021-03-22 22:00:00,ERIC-SAPC,mpsapc1,PL-12,900,12,23
  6 1668k    6  106k    0     0  25.8M      0 --:--:-- --:--:-- --:--:-- 25.8M
curl: (23) Failed writing body (2568 != 16384)

root@8b1a37f7e1d9:/# curl http://ngnix_file_server:80/index.html
<!DOCTYPE html>
<html>
<head>
<title>Welcome to nginx!</title>
<style>
    body {
        width: 35em;
        margin: 0 auto;
        font-family: Tahoma, Verdana, Arial, sans-serif;
    }
</style>
</head>
<body>
<h1>Welcome to nginx!</h1>
<p>If you see this page, the nginx web server is successfully installed and
working. Further configuration is required.</p>

<p>For online documentation and support please refer to
<a href="http://nginx.org/">nginx.org</a>.<br/>
Commercial support is available at
<a href="http://nginx.com/">nginx.com</a>.</p>

<p><em>Thank you for using nginx.</em></p>
</body>
</html>

3) Now that the accessibility is checked when i use the same url http://ngnix_file_server:80/OSProcessingUnit.csv in the dashboard using Infinity plugin still I get the same error

Is it that URL filed supports only https link and not http ? Or still any other missing points for this integration to work ?

![image](https://user-images.githubusercontent.com/67621435/120791361-c1d64500-c551-11eb-81f9-1acf3a64f94c.png)
